core/classes/Util.php

```diff
@@ -611,4 +558,131 @@ public static function replaceAnchorsWithText($data) {
 		}, $data);
 		return $data;
 	}
+
+	public static function getSetting(DB $db, $setting, $fallback = null) {
+		$value = $db->get('settings', array('name', '=', $setting));
+		if ($value->count()) return $value->first()->value;
+		else return $fallback;
+	}
+
+	public static function getDiscordRoleId(DB $db, $group_id) {
+		$discord_role_id = $db->get('group_sync', array('website_group_id', '=', $group_id));
+		if ($discord_role_id->count()) return $discord_role_id->first()->discord_role_id;
+		else return null;
+	}
+
+	public static function getWebsiteGroup(DB $db, $discord_role_id) {
+		$website_group_id = $db->get('group_sync', array('discord_role_id', '=', $discord_role_id));
+		if ($website_group_id->count()) {
+			$group = $db->get('groups', array('id', '=', $website_group_id->first()->website_group_id));
+			if ($group->count()) {
+				$group_array = array();
+				$group_array['group'] = $group->first();
+				$group_array['primary'] = $db->query('SELECT `primary` FROM nl2_group_sync WHERE discord_role_id = ?', array($discord_role_id))->first()->primary;
+				return $group_array;
+			}
+		}
+		return null;
+	}
+	
+	public static function removeDiscordRole($user_query, $group, Language $language) {
+		if (self::getSetting(DB::getInstance(), 'discord_integration')) {
+			// They have a valid discord Id
+			if ($user_query->discord_id != null && $user_query->discord_id != 010) {
+
+				$group_discord_id = self::getDiscordRoleId(DB::getInstance(), $group);
+
+				$api_url = rtrim(self::getSelfURL(), '/') . rtrim(URL::build('/api/v2/' . Output::getClean(self::getSetting(DB::getInstance(), 'mc_api_key')), '', 'non-friendly'), '/');
+
+				// TODO: Probably a nicer way to do this
+				$url = '/roleChange?id=' . $user_query->discord_id . '&guild_id=' . self::getSetting(DB::getInstance(), 'discord');
+
+				if ($group_discord_id == null) return;
+				
+				$url .= '&role=null&oldRole=' . $group_discord_id;
+
+				if ($url != null) {
+					$result = self::discordBotRequest($url . '&api_url=' . $api_url . '/');
+					if ($result != 'success') {
```

Comment:
From here to line 626 feels more like page logic to me, should the function not just return $result and the page does error mapping/redirects

Reply:
I looked into this and since these functions are used only in the user edit page, changing this to page logic does 2 things.
1. Copy/pasting this code to the page reuses it about 4 times, rather than 2 clean functions.
2. Requires more queries in multiple parts of the page to check if discord integration is on.
Overall I think this would be better as it is currently

Reply:
Hmm okay, I just think a method in a class shouldn't dictate what happens with its result, it should be up to whatever calls it to decide what to do. Maybe one to revisit.